### DISCUSSION TITLE
Taxes Provider handle Vat Number

### DISCUSSION THREAD

**Discussion by ricardolopesdev** [medusajs/medusa#10114] [Q&A]
Hello everybody.
I would like to know if anyone knows how to do it at the tax provider to check if the VAT rate is applied based on the VAT identification number.
So far I have managed to add the VAT identification number field to the shipping address through the metadata. And I need it to reach the Tax Provider. Does anyone have an idea how to do it?

**Comment by srindom** [comment]
Is this in v1 or v2?

**Reply by ricardolopesdev** [reply]
It's V2 @srindom

**Comment by jsphpl** [comment]
The [TaxCalculationContext](https://docs.medusajs.com/resources/references/tax/interfaces/tax.TaxCalculationContext), which is the third argument of the tax provider's `getTaxLines()` method, contains an `address` field, which is the shipping address of the cart.

**Reply by ricardolopesdev** [reply]
@jsphpl  Tkanks for the suggestion, but i already implemented the solution for validation for exemption of Taxes when the VAT Number is's Valid.
